ee/modules/110-istio/hooks/discovery_versions_to_install_test.go

```diff
@@ -21,13 +21,34 @@ import (
 var _ = Describe("Istio hooks :: discovery_versions_to_install ::", func() {
 	f := HookExecutionConfigInit(`{"istio":{}}`, "")
 
+	assertNoMetrics := func(f *HookExecutionConfig) {
+		m := f.MetricsCollector.CollectedMetrics()
+		Expect(m).To(HaveLen(0))
+	}
+
+	assertTelemetryMetrics := func(f *HookExecutionConfig, version string) {
+		m := f.MetricsCollector.CollectedMetrics()
+		Expect(m).To(HaveLen(1))
+
+		found := false
+		for _, metric := range m {
```

Comment:
I was reading the loop and then noticed `.To(HaveLen(1))`. Let's just pick `metric := m[0]` after that? It will simplify this code so much

```go
metric := m[0]
Expect(metric.Name).To(Equal("d8_telemetry_istio_control_plane_full_version"))
Expect(metric.Labels["version"]).To(BeEqual(version))
```

Reply:
BTW the naming detail seems kinda reversed to me. I'd go this way:

```go
metrics := f.MetricsCollector.CollectedMetrics()
// ...
m := metrics[0]
```

Reply:
yep, thank you.

done